doc/dev/error-categorization.md

```diff
@@ -0,0 +1,111 @@
+# Error Categorization
+
+In this document we start to explain how to categorize errors.
+Many errors are unique and can be difficult to categorize so this
+document will act as a guideline.
+
+## Categorization Mindset
+
+Errors along with their unique code only exist because they can be **reacted
+differently on programmatically**. Imagine a method call which returns a `Timeout` error. Naturally
+a senseful reaction would be to retry at a later point in time. So
+the reaction here would be to retry with a time based approach. On the other hand
+it does not make sense to differentiate between `No Write Permission` and `No Read Permission` as
+the application just knows that it simply cannot access the desired resource and tells
+the user to grant it. One could argue that an application could always react differently
```

Comment:
> One could argue that an application could always react differently

You provide a counter argument to your own categorisation method, before you even introduced the method... At least move the question above this explanation.